src/opf/utils.jl

```diff
@@ -0,0 +1,67 @@
+function buspair_voltage_bounds(data::OPFData)
+    E = data.E
+    vmin, vmax = data.vmin, data.vmax
+    dvamin, dvamax = data.dvamin, data.dvamax
+    bus_fr, bus_to = data.bus_fr, data.bus_to
+    
+    buspairs = Dict{Tuple{Int,Int},Dict{String,Any}}()
+    for e in 1:E
+        i = bus_fr[e]
+        j = bus_to[e]
+
+        if !haskey(buspairs, (i,j))
+            buspairs[(i,j)] = buspair = Dict{String,Any}()
+            buspair["angmin"] = dvamin[e]
+            buspair["angmax"] = dvamax[e]
+            buspair["edges"] = [e]
+        else
+            buspair = buspairs[(i,j)]
+            buspair["angmin"] = max(buspair["angmin"], dvamin[e])
+            buspair["angmax"] = min(buspair["angmax"], dvamax[e])
+            push!(buspair["edges"], e)
+        end
+    end
```

Comment:
Do we need to go through buspairs? As opposed to iterating over branches?
Pros & cons: 
* buspairs is a little more efficient because we handle all parallel branches at once
* branches would be simpler to read, and match the style of, e.g., the SOC formulation

Reply:
Do you mean not doing the min/max over parallel edges' angmax/angmin?

Reply:
I had not realized that actually :|
To be fair: angle deviation bounds are artificially added in pretty much all the PGLib cases, with a default value of +/- `pi/3` (which is very large). Hence, those constraints will pretty much never be active... so I don't think we loose anything by being simplistic on that front

Reply:
The current `test_voltage_phasor_bounds` is expected to fail for some cases then. It checks for strict equality against PM.

Reply:
Seems to pass with the current test cases